src/Statement/Grouper/FilteringStatementGrouper.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Statement\Grouper;
+
+use InvalidArgumentException;
+use Wikibase\DataModel\Statement\Statement;
+use Wikibase\DataModel\Statement\StatementFilter;
+use Wikibase\DataModel\Statement\StatementList;
+
+/**
+ * @since 3.2
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+class FilteringStatementGrouper implements StatementGrouper {
```

Comment:
There **are** groupers that don't work with filters. This is one of the main reasons why I added the most basic groupers to this patch, even if they are not relevant for the current story. Also see https://github.com/wmde/WikibaseDataModelServices/pull/91/files#r46273418.